libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -675,144 +693,193 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
+
+        /**
+         * For QR code login URLs, the Salesforce Identity API UI Bridge parameters JSON key for the
+         * PKCE code verifier, which is only used when the front door URL was generated for the web
+         * server authorization flow.  The user agent flow does not require a value for this
+         * parameter.
+         *
+         * Apps may customize this so long as it matches the server-side Apex class or other code
+         * generating the QR code.
+         */
+        @Suppress("MemberVisibilityCanBePrivate")
+        var qrCodeLoginUrlJsonPkceCodeVerifierKey = "pkce_code_verifier"
+
+        /**
+         * When QR code log in is enabled, determines if the provided intent has QR code login
+         * parameters.
+         * @param intent The intent to determine QR code login enablement for
+         * @return Boolean true if the intent has QR code login parameters or false otherwise
+         */
+        fun isQrCodeLoginIntent(
+            intent: Intent
+        ) = SalesforceSDKManager.getInstance().isQrCodeLoginEnabled
+                && intent.data?.path?.contains(qrCodeLoginUrlPath) == true
+
+        /**
+         * Parses Salesforce Identity API UI Bridge parameters from the provided login QR code login
+         * URL.
+         * @param qrCodeLoginUrl The QR code login URL
+         * @return The UI Bridge API parameters or null if the QR code login URL cannot provide them
+         * for any reason
+         */
+        fun uiBridgeApiParametersFromQrCodeLoginUrl(
+            qrCodeLoginUrl: String?
+        ) = qrCodeLoginUrl?.let { qrCodeLoginUrlUnwrapped ->
+            uiBridgeApiJsonFromQrCodeLoginUrl(qrCodeLoginUrlUnwrapped)?.let { uiBridgeApiJson ->
+                uiBridgeApiParametersFromUiBridgeApiJson(uiBridgeApiJson)
+            }
+        }
 
-        /** The QR code login intent path */
-        private const val LOGIN_QR_PATH = "/login/qr"
+        /**
+         * A data class representing Salesforce Identity API UI Bridge parameters.
+         */
+        data class UiBridgeApiParameters(
 
-        /** The login QR code's UI Bridge API parameter's JSON frontdoor bridge URL key */
-        private const val FRONTDOOR_BRIDGE_URL_KEY = "frontdoor_bridge_url"
+            /** The front door bridge URL */
+            val frontdoorBridgeUrl: String,
+
+            /** The PKCE code verifier */
+            val pkceCodeVerifier: String?
+        )
 
-        /** The login QR code's UI Bridge API parameter's JSON PKCE code verifier key */
-        private const val PKCE_CODE_VERIFIER_KEY = "pkce_code_verifier"
+        // endregion
+        // region QR Code Login Via Salesforce Identity API UI Bridge Private Implementation
+
+        /**
+         * For QR code login URLs, a regular expression to extract the Salesforce Identity API UI
+         * Bridge parameter JSON string.
+         */
+        private val qrCodeLoginJsonRegex by lazy {
```

Comment:
@wmathurin - I couldn't reproduce the "external" regular expression you documented in the spike.  My internal and external QR code URLs are identical on a Google Pixel 9 Pro.